mailet/standard/src/test/java/org/apache/james/transport/matchers/AtMostTest.java

```diff
@@ -56,103 +57,208 @@ private Mail createMail() throws MessagingException {
+                .build()))
+                .isInstanceOf(MessagingException.class);
+        }
+
+        @Test
+        void shouldThrowWhenMatchersConfigWithoutConditionName() {
+            assertThatThrownBy(() -> new AtMost().init(FakeMatcherConfig.builder()
+                .matcherName("NoValueMatcher")
+                .condition(":3")
+                .build()))
+                .isInstanceOf(IllegalArgumentException.class);
+        }
+
+        @Test
+        void shouldThrowWhenMatchersConfigNameAsSpace() {
+            assertThatThrownBy(() -> new AtMost().init(FakeMatcherConfig.builder()
+                .matcherName("NoValueMatcher")
+                .condition("  :  ")
+                .build()))
+                .isInstanceOf(MessagingException.class);
+        }
     }
 
-    @Test
-    void shouldMatchWhenNoRetries() throws MessagingException {
-        Mail mail = createMail();
-        mail.setAttribute(new Attribute(AT_MOST_TRIES, AttributeValue.of(0)));
+    @Nested
+    class MultiplesMatcherConfigurationTest {
+        private AtMost multipleMatchers;
 
-        Collection<MailAddress> actual = matcher.match(mail);
+        @BeforeEach
+        void setup() throws MessagingException {
+            this.multipleMatchers = new AtMost();
+            multipleMatchers.init(
+                FakeMatcherConfig.builder()
+                    .matcherName("MultipleMatchers")
+                    .condition("randomName:3")
+                    .condition("randomName:5")
+                    .build());
+        }
 
-        assertThat(actual).containsOnly(RECIPIENT1);
-    }
+        @Test
+        void matchersShouldMatchWhenNoRetries() throws MessagingException {
+            Mail mail = createMail();
+            mail.setAttribute(new Attribute(AttributeName.of("MultipleMatchers"), AttributeValue.of(0)));
 
-    @Test
-    void shouldNotMatchWhenOverAtMost() throws MessagingException {
-        Mail mail = createMail();
-        mail.setAttribute(new Attribute(AT_MOST_TRIES, AttributeValue.of(3)));
+            Collection<MailAddress> actual = multipleMatchers.match(mail);
 
-        Collection<MailAddress> actual = matcher.match(mail);
+            assertThat(actual).containsOnly(RECIPIENT1);
+        }
 
-        assertThat(actual).isEmpty();
-    }
+        @Test
+        void matchersShouldStopWhenAMatcherReachedLimit() throws MessagingException {
+            Mail mail = createMail();
+            mail.setAttribute(new Attribute(AttributeName.of("MultipleMatchers"), AttributeValue.of(3)));
 
-    @Test
-    void shouldNotMatchWhenEqualToAtMost() throws MessagingException {
-        Mail mail = createMail();
-        mail.setAttribute(new Attribute(AT_MOST_TRIES, AttributeValue.of(2)));
+            Collection<MailAddress> actual = multipleMatchers.match(mail);
 
-        Collection<MailAddress> actual = matcher.match(mail);
+            assertThat(actual).containsOnly(RECIPIENT1);
```

Comment:
It matches then, so your test is wrong (it says it should not). Easy to explain with previous comment on your setup